src/test/java/com/photolocos/enterprise/EnterpriseApplicationTests.java

```diff
@@ -2,62 +2,92 @@
 
 import com.photolocos.enterprise.dao.ILocationDAO;
 import com.photolocos.enterprise.dao.IPhotoDAO;
-import com.photolocos.enterprise.dao.PhotoDAO;
-import com.photolocos.enterprise.dto.PhotoDTO;
+import com.photolocos.enterprise.dto.Location;
+import com.photolocos.enterprise.dto.Photo;
 import org.junit.jupiter.api.Test;
-import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.util.Assert;
 
-import java.util.ArrayList;
-import java.util.List;
+import java.util.Arrays;
 import java.util.Set;
 
-import static com.photolocos.enterprise.dao.PhotoDAO.*;
-
 @SpringBootTest
 class EnterpriseApplicationTests {
 
-    private PhotoDTO photo = new PhotoDTO();
-
-    @MockBean
+    @Autowired
     private ILocationDAO locationDAO;
+    @Autowired
     private IPhotoDAO photoDAO;
 
 
     @Test
     void contextLoads() {
     }
 
+    @Test
+    void userSearchesLocation_returnsPhotoResults() throws Exception {
+        givenPhotoWithStateLocationExists();
+        returnsPhotoResults(userSearchesLocation());
+    }
+
+    private void givenPhotoWithStateLocationExists() throws Exception {
+        Photo photo = new Photo();
+        Location location = new Location();
+        location.setState("Ohio");
+
+        photo.setLocation(location);
+        photoDAO.createEntry(photo);
+    }
+
+    private Set<Photo> userSearchesLocation() {
+        return photoDAO.fetchByLocation("Ohio");
+    }
+
+    private void returnsPhotoResults(Set<Photo> results) {
+        Assert.notNull(results, "Photo search results returned null");
+        Assert.notEmpty(results, "Photo search results returned empty set.");
+
+        boolean locationMatches = true;
+        for(Photo photo: results){
+            if(!photo.getLocation().getState().equals("Ohio")){
+                locationMatches = false;
+                break;
+            }
+        }
+
+        Assert.isTrue(locationMatches, "Photo search returned an incorrect result set.");
+    }
+
```

Comment:
There was only one unit test before and it wasn't passing. Here I added a unit test based on one of the use cases in your read me file. [(SonarQube Seven Deadly Sins: Lack of Unit Tests)](https://docs.sonarqube.org/display/SONARQUBE45/Lack+of+Unit+Tests)